src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -132,6 +145,16 @@ public void deleteRoles(final Node node) throws RepositoryException {
                 rdfTypes.addAll(x.getTypes());
             });
 
+        // If we get the default Authorization and it contains acl:accessTo properties,
```

Comment:
This block handles the following case:

Say you make a request for /rest/foo/bar/baz

If there is no acl:accessControl triple on that resource, it searches up the hierarchy to the root. If no acl:accessControl triple is found (`effectiveAcl.isPresent() == false`), the filesystem backstop is checked. Now, say that filesystem backstop contains an acl:accessTo triple pointing to the root: acl:accessTo <info:fedora/> (or, alternately: acl:accessTo <info:fedora/foo>). This is where this block comes in.

Later on, when the code checks the `List<WebACAuthorization>`, it needs to make sure that the path of the requested resource matches the resource pointed to by acl:accessTo, but if the filesystem backstop is used, we need a way to line up the path of the original resource and whatever is listed in the acl:Authorization. In the more traditional case (where there is an acl:accessControl triple somewhere in the hierarchy), we simply use that path, but we don't have that information in this case. Thus, we need to provide all the possible paths: /rest/, /rest/foo, /rest/foo/bar and /rest/foo/baz. That is what this block generates. With that list in place, it can be compared to the acl:accessControl triple(s) in the filesystem backstop. If _any_ of those match, then the ACL modes are applied. If _none_ of them match (and there is no relevant acl:accessToClass), then the request is denied.

The block is located here because `resourcePaths` is needed in the next line when the `checkAccessTo` Predicate is created. (The `accessTo` function forms a closure over the value of `resourcePaths`).